services/property-svc/internal/property/queries/v1/get_property_by_id.go

```diff
@@ -66,6 +73,12 @@ func NewGetPropertyByIDQueryHandler() GetPropertyByIDQueryHandler {
 			}
 		}
 
+		// Verify user is allowed to see this property
+		check := hwauthz.NewPermissionCheck(user, perm.PropertyCanUserGet, perm.Property(propertyID))
+		if err = authz.Must(ctx, check); err != nil {
+			return nil, 0, err
+		}
+
```

Comment:
Move this to the start of the command. We only spend computing if the checks succeeds.